packages/rum/test/userActionCollection.spec.ts

```diff
@@ -92,9 +88,48 @@ describe('startUserActionCollection', () => {
     button.click()
 
     clock.expire()
+  }
+
+  it('cancels pending user action on view loading', () => {
+    const fakeLocation: Partial<Location> = { pathname: '/foo' }
+    const fakeSession = {
+      getId() {
+        return '42'
+      },
+    }
+
+    button.addEventListener(DOM_EVENT.CLICK, () => {
+      clock.tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+      // Since we don't collect dom mutations for this test, manually dispatch one
+      lifeCycle.notify(LifeCycleEventType.DOM_MUTATED)
+    })
+
+    clock.tick(SOME_ARBITRARY_DELAY)
+    button.click()
+    lifeCycle.notify(LifeCycleEventType.VIEW_COLLECTED, {
+      documentVersion: 0,
+      duration: 0,
+      id: 'foo',
+      loadingType: ViewLoadingType.INITIAL_LOAD,
+      location: fakeLocation as Location,
+      measures: {
+        errorCount: 0,
+        longTaskCount: 0,
+        resourceCount: 0,
+        userActionCount: 0,
+      },
+      startTime: 0,
+    })
```

Comment:
Since we don't care about the content of the view in this test, what about using a stub:
```ts
const fakeView = {}
lifeCycle.notify(LifeCycleEventType.VIEW_COLLECTED, fakeView as View)
```
It would remove unnecessary noise and avoid that this test break when there is a change in the view event structure